zipline-loader/src/commonMain/kotlin/app/cash/zipline/loader/LoadResult.kt

```diff
@@ -18,9 +18,22 @@ package app.cash.zipline.loader
 import app.cash.zipline.Zipline
 
 /**
- * Zipline engine loaded with code and related metadata on the load attempt.
+ * Final result state from a ZiplineLoader attempt.
  */
-data class LoadedZipline(
-  val zipline: Zipline,
-  val freshAtEpochMs: Long,
-)
+sealed class LoadResult {
+  /**
+   * [Zipline] engine successfully loaded with code and corresponding metadata.
+   */
+  data class Success(
+    val zipline: Zipline,
+    val freshAtEpochMs: Long,
```

Comment:
Not sure what this is used for, but could be an `abstract val` on the `LoadResult` type if we think there's code that would unconditionally want to access it.

Reply:
Maybe even simpler - omit the `freshAtEpochMs` for failure, since that’s not particularly useful? The main utility of this thing is to know when you’re reading stale code from the cache, but it’s not relevant that a failure is fresh or stale.